packages/autocertifier-client/src/createSelfSignedCertificate.ts

```diff
@@ -0,0 +1,79 @@
+import * as forge from 'node-forge'
+
+interface Certificates {
+    caCert: string
+    caKey: string
+    serverCert: string
+    serverKey: string
+}
+
+export function createSelfSignedCertificate(fqdn: string, validMonths: number, validMilliseconds?: number): Certificates {
+    // Generate a new RSA key pair for the certificate authority
+    const caKeys = forge.pki.rsa.generateKeyPair(2048)
+
+    // Create a new X.509 certificate for the certificate authority
+    const caCert = forge.pki.createCertificate()
+    caCert.publicKey = caKeys.publicKey
+    caCert.serialNumber = '01'
```

Comment:
Is this q magic constant of just number?

Reply:
It is just a number given for the self signed certificate. The value does not matter in this case